### PR TITLE
Remove file-per-thread-logger support from wasmtime

### PR DESCRIPTION
**Pull request by alexcrichton**
I was a bit surprised today when configuring `RUST_LOG` to get some
output from wasmtime that it started creating a lot of log files in the
current directory. This looks to be the behavior of the
`file-per-thread-logger` crate and at least intentional. There's also a
`-d` flag to the `wasmtime` CLI to to switch to one that prints on
stderr.

This looks like it was added a long time ago b3d831015c so I'd like to
propose that we remove the logger. There's also a good deal of
management of how it's initialized which has seemed to baloon quite a
bit over time, and I'm not sure all of the initialization is necessarily
pulling its weight anymore either.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -88,9 +62,9 @@ struct CommonOptions {
     #[structopt(long, conflicts_with = "lightbeam")]
     cranelift: bool,
 
-    /// Enable debug output
-    #[structopt(short, long)]
-    debug: bool,
+    /// Enable debug output, currently unused
+    #[structopt(short = "d", long = "debug")]
+    _debug: bool,
```
Comment (peterhuene) [inline src/lib.rs]: Perhaps we should make the flag now set `RUST_LOG` (if not set)?

**Comment by sunfishcode** [comment]
Some parts of Wasmtime are threaded, and in the future more parts will be. With the usual single-file loggers, the log messages from multiple threads are mostly unusable. One option is to limit thread pool sizes to 1 when debugging, which sometimes works, but makes it difficult to debug problems that are connected to the threading in some way. Can you say more about how you expect logging to be used?

**Comment by alexcrichton** [comment]
At least when working on other projects I've never come across the need to isolate the logs of each thread into its own file. When I've done some debugging I've typically found it more useful to have all the output interleaved so I can figure out what else is going on at the same time as well. Otherwise though I figure we could use our own logger which prepends a thread-specific prefix (but doesn't require such intrusive management for us to name all the threads) or the logging would have a prefix you could grep for to see what you're interested in.

File: src/lib.rs (hunk 1 above)
Comment (peterhuene) [reply src/lib.rs]: If `wasmtime=debug` doesn't offer anything useful, I would argue just for `debug` and let users drink from the fire hose; it's always possible for them to tailor it by setting `RUST_LOG` manually.

**Comment by sunfishcode** [comment]
Cranelift currently does emit some multi-line log messages, for IR dumps, which aren't very greppable. We could probably rework things so that it emits a log message per line, although if the lines have prefixes it's yet another step developers need to do to get an IR file out that they can test with.

Would it be better if we emitted IR dumps somewhere outside of the logging mechanism?

**Comment by alexcrichton** [comment]
Hm ok, I figured this'd be an easy cleanup but if it's serving a purpose then it's serving a purpose so it shouldn't be removed. I find the behavior of `RUST_LOG=foo wasmtime foo.wasm` spewing a lot of log files too surprising to be comfortable with, but we can figure something else out for that later if someone is motivated enough.

**Comment by sunfishcode** [comment]
One easy thing we could do would be to get rid of `-d` and have wasmtime spew to stderr by default when you give it a RUST_LOG=foo, and have a non-default flag for enabling file-per-thread logging.

**Pull request closed without merging**